src/sentry/tasks/split_discover_dataset.py

```diff
@@ -0,0 +1,226 @@
+            sentry_sdk.capture_exception(e)
+        except Exception as e:
+            sentry_sdk.capture_exception(e)
+
+
+def update_widget_discover_split(widget: DashboardWidget):
+    if (
+        widget.widget_type != DashboardWidgetTypes.DISCOVER
+        or widget.discover_widget_split is not None
+    ):
+        metrics.incr("task.split_discover_dataset.skip", tags={"status": "skipped"})
+        return
+
+    now = datetime.datetime.now()
+    yesterday = now - datetime.timedelta(days=1)
+    params: ParamsType = {
+        "project_objects": widget.dashboard.projects.all(),
+        "start": yesterday,
+        "end": now,
+        "organization_id": widget.dashboard.organization.id,
+    }
+
+    original_data = get_widget_data(widget, params)
+    error_data = get_widget_data(widget, params, event_type="error")
+    transaction_data = get_widget_data(widget, params, event_type="transaction")
+
+    error_data_match, transaction_data_match = compare_results(
+        original_data, error_data, transaction_data
+    )
+
+    split = get_split_decision(error_data_match, transaction_data_match)
+    logger.log(logging.INFO, "Split decision for widget %s: %s", widget.id, split)
+    metrics.incr("task.split_discover_dataset.update", tags={"split": split})
+
+    widget.update(discover_widget_split=split)
+
+
+def get_widget_data(widget: DashboardWidget, params: ParamsType, event_type: str | None = None):
+    widget_query = DashboardWidgetQuery.objects.filter(widget_id=widget).order_by("id").first()
+
+    conditions = widget_query.conditions
+    if event_type:
+        if conditions:
+            conditions = f"({widget_query.conditions}) AND event.type:{event_type}"
+        else:
+            conditions = f"event.type:{event_type}"
+
+    query_builder = QueryBuilder(
+        dataset=Dataset.Discover,
+        params=params,
+        selected_columns=widget_query.aggregates + widget_query.columns,
+        query=conditions,
+        config=QueryBuilderConfig(
+            transform_alias_to_input_format=True,
+        ),
+    )
+
+    results = query_builder.run_query(Referrer.METRIC_EXTRACTION_SPLIT_DISCOVER_DATASET.value)
+    processed_results = query_builder.process_results(results)
+
+    return processed_results
+
+
+def compare_results(original_data, error_data, transaction_data) -> tuple[bool, bool]:
+    threshold = options.get("split_discover_dataset.data_distance_threshold")
+
+    error_data_match, transaction_data_match = True, True
+    for (o, e, t) in zip(original_data["data"], error_data["data"], transaction_data["data"]):
+        for column in o:
+            if not is_datapoint_close(o[column], e.get(column, float("inf")), threshold):
+                error_data_match = False
+            if not is_datapoint_close(o[column], t.get(column, float("inf")), threshold):
+                transaction_data_match = False
+
+    return error_data_match, transaction_data_match
```

Comment:
The only problem with this I can see is that we are running the query for the last day, but not all widgets are setup that way. 

eg. a lot of people make investigative dashboards where they look for specific issues that are infrequent or they have widgets setup for outlier errors, if we're solely doing `not is_datapoint_close` then for the last day (where there are no data points) we'll get `a == b` which means it'll say `error_data_match = True`, `transaction_data_match = True` whereas really if we have both `a=0` and `b=0` it means the result is ambiguous.

An ambiguous result we'll likely still want to fall back to "we're not sure" w/ discover or use `decision=None` but regardless we should at least track that as a metric vs. successfully picking w/ data to know how many of our `discover` column values so we can see if something seems amiss or we can change the period if we're getting too many of these values etc.